### PR TITLE
`povm_toolbox.library` documentation clean up

### PR DESCRIPTION
**Pull request by mrossinek**
This goes through ~all modules~ the `povm_toolbox.library` module and cleans up its documentation.
In doing so, I am also re-evaluating whether each attribute/method should be public or private, thereby tackling the last task of #35 (for this module).

### CODE REVIEW

**Comment by mrossinek** [comment]
As per my updates to the title and PR description, I am going to limit this PR to the library module in order to keep it somewhat reasonably sized.
I will follow up with more PRs for the other modules.

File: povm_toolbox/library/locally_biased_classical_shadows.py [hunk 1]
```diff
@@ -8,61 +8,75 @@
...
 from numpy.random import Generator
 
-from .randomized_projective_measurements import RandomizedProjectiveMeasurements
+from .mutually_unbiased_bases_measurements import MutuallyUnbiasedBasesMeasurements
 
 
-class LocallyBiasedClassicalShadows(RandomizedProjectiveMeasurements):
-    """A locally-biased classical shadows POVM."""
+class LocallyBiasedClassicalShadows(MutuallyUnbiasedBasesMeasurements):
+    """A locally-biased classical shadows POVM.
```
Comment (timmintam) [inline povm_toolbox/library/locally_biased_classical_shadows.py:22]: Nice, good idea to make it a subclass of `MutuallyUnbiasedBasesMeasurementa` now that we have it!

File: povm_toolbox/library/povm_implementation.py [hunk 2]
```diff
@@ -182,15 +227,34 @@ def _povm_outcomes(
         povm_metadata: MetadataT,
         loc: int | tuple[int, ...] | None = None,
     ) -> list[tuple[int, ...]]:
-        """TODO."""
+        """Convert the raw bitstrings into POVM outcomes based on the associated metadata.
+
+        Args:
+            bit_array: The raw bitstrings.
+            povm_metadata: The associated metadata.
+            loc: an optional location to splice the bitstrings.
```
Comment (timmintam) [inline povm_toolbox/library/povm_implementation.py:235]: I am not familiar with the term "splice", was it what you meant ? Or "slice" ?

Comment (timmintam) [reply povm_toolbox/library/povm_implementation.py:235]: (more occurences below as well)

File: povm_toolbox/library/randomized_projective_measurements.py [hunk 3]
```diff
@@ -193,35 +248,33 @@ def to_sampler_pub(
...
-            shots: A specific number of shots to run with. Note that ``shots`` is
-                effectively the number of times we sample PVMs. Then, for each
-                sampled PVM, the measurement is repeated ``self.shot_repetitions``
-                times. Therefore, the actual total number of measurement shots is
-                then ``shots`` multiplied by ``self.shot_repetitions``.
-            pass_manager: An optional pass manager. After the supplied circuit has
-                been composed with the measurement circuit, the pass manager will
+            shots: A specific number of shots to run with.
+            .. note::
+               test
```
Comment (timmintam) [inline povm_toolbox/library/randomized_projective_measurements.py:269]: "note" to be completed ?

**Review by timmintam — COMMENTED** [review]
Really nice, looks good! Only 1-2 small comments

File: povm_toolbox/library/povm_implementation.py (hunk 2 above)
Comment (mrossinek) [reply povm_toolbox/library/povm_implementation.py:235]: Ah yeh that was a typo. Splicing exists, too, but not in this context. Will change it right away

File: povm_toolbox/library/randomized_projective_measurements.py (hunk 3 above)
Comment (mrossinek) [reply povm_toolbox/library/randomized_projective_measurements.py:269]: Oh oops :smile: I was testing whether I could include a `.. note` admonition here to document the relation of `shots` and `shot_repetitions` but I ended up moving that higher up in the docstring

**Review by timmintam — APPROVED** [review]

**Pull request merged**